### PR TITLE
feat: add fundraising banners across homepage, navigation drawer, and quran reader with analytics   

### PR DESCRIPTION
**Pull request by yousefdergham**
## Summary

Adds the `HomepageFundraisingBanner` to three placements (homepage, navigation drawer, Quran reader) and instruments every user interaction with self-descriptive analytics events. Also fixes the Redux dismiss-state so that users who dismissed last year's banner will see this year's campaign.

- **Banner placement**: homepage (existing), navigation drawer (new), end-of-surah / end-of-juz/page/range/hizb/rub sections in the Quran reader (new)
- **Analytics**: 30 uniquely named events across all placements, each carrying a context param (chapter ID, juz number, page number, verse key, etc.)
- **Share modal**: removed the "Generate Media" (Quran Reader) option from the fundraising share popup — not relevant to a donation share flow
- **Navbar banner**: updated CTA text and link to point to the external donation page
- **Redux migration v46**: resets `isHomepageBannerVisible` and `isQuranReaderBannerVisible` to `true` so returning users who dismissed last Ramadan's banner will see this year's campaign
---

## Homepage & Navigation
| Placement | Donate | Share | Learn More | Dismiss | Params |
|---|---|---|---|---|---|
| Homepage | `homepage_donation_section_donate_clicked` | `homepage_donation_section_share_clicked` | `homepage_donation_section_learn_more_clicked` | `homepage_donation_section_dismissed_clicked` | — |
| Burger menu | `navigation_drawer_fundraising_banner_donate_clicked` | — | `navigation_drawer_fundraising_banner_learn_more_clicked` | — | — |
| Navbar header banner | `donate_button_banner_clicked` | — | — | — | — |

---

## Quran Reader
| Placement | Donate | Share | Learn More | Dismiss | Params |
|---|---|---|---|---|---|
| Chapter | `quran_reader_chapter_donate_clicked` | `quran_reader_chapter_share_clicked` | `quran_reader_chapter_learn_more_clicked` | `quran_reader_chapter_dismissed_clicked` | `{ chapterId: N }` |
| Juz | `quran_reader_juz_donate_clicked` | `quran_reader_juz_share_clicked` | `quran_reader_juz_learn_more_clicked` | `quran_reader_juz_dismissed_clicked` | `{ juzNumber: N }` |
| Page | `quran_reader_page_donate_clicked` | `quran_reader_page_share_clicked` | `quran_reader_page_learn_more_clicked` | `quran_reader_page_dismissed_clicked` | `{ pageNumber: N }` |
| Verse/Range | `quran_reader_range_donate_clicked` | `quran_reader_range_share_clicked` | `quran_reader_range_learn_more_clicked` | `quran_reader_range_dismissed_clicked` | `{ verseKey: "2:255" }` |
| Hizb | `quran_reader_hizb_donate_clicked` | `quran_reader_hizb_share_clicked` | `quran_reader_hizb_learn_more_clicked` | `quran_reader_hizb_dismissed_clicked` | `{ hizbNumber: N }` |
| Rub | `quran_reader_rub_donate_clicked` | `quran_reader_rub_share_clicked` | `quran_reader_rub_learn_more_clicked` | `quran_reader_rub_dismissed_clicked` | `{ rubNumber: N }` |

### Pre-existing events that continue to fire (unchanged)

| Placement | Event Name | Params |
|---|---|---|
| Burger menu — Donate button | `donate_button_clicked` | `{ source: "sidebar_banner_monthly" }` |

---
## Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [ ] ♻️ Refactoring (no functional changes)

## Scope Confirmation

- [x] This PR addresses **one** feature/fix only
- [x] If multiple changes were needed, they are split into separate PRs

## Environment Variables

N/A

## Rollback Safety

- [x] Can be safely reverted without data issues or migrations

> If reverted after the v46 migration has already run for a user, their banner dismiss state will be reset to visible — benign, they'll just see the banner once more.

## Test Plan

- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [x] Manual testing performed

**Testing steps:**

1. Visit the homepage — fundraising banner should be visible; verify donate/share/learn-more/dismiss fire the correct `homepage_donation_section_*_clicked` events
2. Open the navigation drawer — fundraising banner should appear; verify donate fires both `donate_button_clicked` (source: `sidebar_banner_monthly`) and `navigation_drawer_fundraising_banner_donate_clicked`; learn-more fires `navigation_drawer_fundraising_banner_learn_more_clicked`
3. Read a surah to the end — banner should appear below end-of-surah section; events should be `quran_reader_chapter_*_clicked` with `{ chapterId: N }`
4. Navigate to a juz/page/hizb/rub/verse URL — banner should appear; events should be `quran_reader_{type}_*_clicked` with the appropriate context param
5. Open the share modal on the fundraising banner — confirm "Generate Media" button is **not** present
6. Dismiss the banner, reload — should stay dismissed (Redux persist). Clear localStorage, reload — should reappear
7. Simulate a returning user who dismissed last year (manually set `isHomepageBannerVisible: false` in localStorage at store version < 46) — after load the migration should reset it to `true`

### Edge Cases Verified

- [x] ⏳ Loading state handled — banner only renders after Redux rehydration (redux-persist gate)
- [ ] ❌ Error state handled
- [ ] 📭 Empty state handled
- [x] 👤 Logged-in vs guest behavior — dismiss state is localStorage-based (same for both); no server sync needed

### Code Quality

- [x] I have performed a **self-review** of my code (file by file)
- [x] My code follows the project style guidelines
- [ ] No `any` types used (or justified if unavoidable) — `analyticsParams?: Record<string, any>` uses `any`; justified because analytics param values are inherently mixed-type
- [x] No unused code, imports, or dead code included
- [x] Complex logic has inline comments explaining "why"
- [x] Functions are under 30 lines and follow single responsibility

### Testing & Validation

- [ ] All tests pass locally (`yarn test`)
- [x] Linting passes (`yarn lint`)
- [ ] Build succeeds (`yarn build`)
- [x] Edge cases and error scenarios are handled

### Localization (if UI changes)

- [x] All user-facing text uses `next-translate`
- [x] Only English locale files modified (Lokalise handles others)
- [ ] RTL layout verified

### Accessibility (if UI changes)

- [x] Semantic HTML elements used
- [x] ARIA attributes added where needed
- [x] Keyboard navigation works

## Reviewer Notes

- The `analyticsSource` + `analyticsParams` props on `HomepageFundraisingBanner` make it reusable for future placements without touching event-logging logic inside the component
- Redux store version bumped from 45 → 46 with a migration that resets banner visibility — this is intentional to re-surface the banner for Ramadan 2026
- Burger menu donate fires two events: the generic `donate_button_clicked` (from `DonateButton` internals, unchanged) plus the new placement-specific `navigation_drawer_fundraising_banner_donate_clicked`

## AI Assistance Disclosure

- [ ] AI tools were NOT used for this PR
- [x] AI tools were used, and I have **thoroughly reviewed and validated** all generated code

### CODE REVIEW

File: src/components/HomePage/ReadingSection/NewCard/index.tsx [hunk 1]
```diff
@@ -1,78 +1,41 @@
...
+            <span aria-hidden="true">✨</span>
+            <span className={styles.srOnly}>Sparkles icon</span>
           </div>
-          <IconContainer
-            className={styles.arrowIcon}
-            size={IconSize.Xsmall}
-            icon={<ArrowIcon />}
-            shouldForceSetColors={false}
-          />
+          <div className={styles.link}>Join the Ramadan Challenge</div>
```
[the remark this answers is not in the document]
Comment (yousefdergham) [reply src/components/HomePage/ReadingSection/NewCard/index.tsx]: good catch

File: src/components/HomePage/ReadingSection/NewCard/index.tsx [hunk 2]
```diff
@@ -1,78 +1,41 @@
...
+        <IconContainer
+          className={styles.arrowIcon}
+          size={IconSize.Xsmall}
+          icon={<ArrowIcon />}
+          shouldForceSetColors={false}
+        />
+      </div>
+    </Card>
   );
 };
```
[the remark this answers is not in the document]
Comment (yousefdergham) [reply src/components/HomePage/ReadingSection/NewCard/index.tsx]: this is realted
